cast/src/lib.rs

```diff
@@ -1077,6 +1080,85 @@ impl SimpleCast {
+    ///    assert_eq!(Cast::to_rlp("[]".to_string()).unwrap(), "0xc0");
+    ///    assert_eq!(Cast::to_rlp("[\"a\"]".to_string()).unwrap(), "0xc161");
+    ///    assert_eq!(Cast::to_rlp("[\"a\",\"b\"]".to_string()).unwrap(), "0xc26162");
+    ///     Ok(())
+    /// }
+    /// ```
+    pub fn to_rlp(value: String) -> Result<String> {
+        let val = serde_json::from_str(&value)?;
+        let item = rlp_converter::value_to_item(&val, false);
+        Ok(format!("0x{}", hex::encode(rlp::encode(&item))))
+    }
+
+    /// Decodes rlp encoded hexadecimal string
+    ///
+    /// ```
+    /// use cast::SimpleCast as Cast;
+    ///
+    /// fn main() -> eyre::Result<()> {
+    ///     assert_eq!(Cast::from_rlp("0x87666f756e647279".to_string()).unwrap(), "\"foundry\"");
+    ///     assert_eq!(Cast::from_rlp("87666f756e647279".to_string()).unwrap(), "\"foundry\"");
+    ///     assert_eq!(Cast::from_rlp("0xc0".to_string()).unwrap(), "[]");
+    ///     assert_eq!(Cast::from_rlp("c161".to_string()).unwrap(), "[\"a\"]");
+    ///     assert_eq!(Cast::from_rlp("0xc26162".to_string()).unwrap(), "[\"a\",\"b\"]");
+    ///     Ok(())
+    /// }
+    /// ```
+    pub fn from_rlp(value: String) -> Result<String> {
+        let striped_value = strip_0x(&value);
+        let bytes = hex::decode(striped_value).expect("could not decode hex");
+        Ok(format!("{}", rlp::decode::<Item>(&bytes).expect("rlp decoding failed")))
+    }
+
+    /// Encodes hex data or list of hex data to hexadecimal rlp
+    ///
+    /// ```
+    /// use cast::SimpleCast as Cast;
+    ///
+    /// fn main() -> eyre::Result<()> {
+    ///     assert_eq!(Cast::hex_to_rlp("[]").unwrap(),"0xc0".to_string());
+    ///     assert_eq!(Cast::hex_to_rlp("0x22").unwrap(),"0x22".to_string());
+    ///     assert_eq!(Cast::hex_to_rlp("[\"0x61\"]",).unwrap(), "0xc161".to_string());
+    ///     assert_eq!(Cast::hex_to_rlp( "[\"0x61\",\"62\"]").unwrap(), "0xc26162".to_string());
+    ///     Ok(())
+    /// }
+    /// ```
+    pub fn hex_to_rlp(value: &str) -> Result<String> {
+        let val = serde_json::from_str(value).unwrap_or(Value::String(value.parse()?));
+        let item = rlp_converter::value_to_item(&val, true);
+        Ok(format!("0x{}", hex::encode(rlp::encode(&item))))
+    }
+
+    /// Decodes rlp encoded list with hex data
+    ///
+    /// ```
+    /// use cast::SimpleCast as Cast;
+    ///
+    /// fn main() -> eyre::Result<()> {
+    ///     assert_eq!(Cast::hex_from_rlp("0xc0".to_string()).unwrap(), "[]");
+    ///     assert_eq!(Cast::hex_from_rlp("0x0f".to_string()).unwrap(), "\"0x0f\"");
+    ///     assert_eq!(Cast::hex_from_rlp("0x33".to_string()).unwrap(), "\"0x33\"");
+    ///     assert_eq!(Cast::hex_from_rlp("0xc161".to_string()).unwrap(), "[\"0x61\"]");
+    ///     assert_eq!(Cast::hex_from_rlp("0xc26162".to_string()).unwrap(), "[\"0x61\",\"0x62\"]");
+    ///     Ok(())
+    /// }
+    /// ```
+    pub fn hex_from_rlp(value: String) -> Result<String> {
```

Comment:
What's the difference between this and `from_rlp`?

Reply:
`from_rlp` tries to display the data as strings while `hex_from_rlp` diplays the hex content, but i guess only the hex version has real use cases.

Should I delete `from_rlp` and replace it by `hex_from_rlp` ?

Reply:
I'm not sure - is decoding the fields as a string always accurate?

Reply:
Decoding as a string can fail but most importantly i don't think data in rlp is likely to be only strings, looking at usual uses in ethers we mostly decode ints or bytes, occasionally strings but trying to decode all fields as strings is useless.
I'm gonna keep only the from hex version as `from-rlp`

@tynes don't hesitate to weigh in as you requested the feature :+1:

Reply:
I think the hex version is more general and useful, if we want to parse specific fields as strings perhaps the output could be piped to `jq` and then `cast --to-ascii`?